examples/firebase/index.js

```diff
@@ -0,0 +1,179 @@
+                    newPass.primaryFields.push({
+                        key: `primary${index}`,
+                        label: field.label,
+                        value: (currentPassType == "boardingPass") ? field.value.toUpperCase() : field.value,
+                    });
+                }
+                });
+
+            request.body.secondary.forEach((field, index) => {
+            if (field.label !== '' || field.value !== '') {
+                newPass.secondaryFields.push({
+                    key: `secondary${index}`,
+                    label: field.label,
+                    value: field.value,
+                    textAlignment: (index === request.body.secondary.length - 2 || index === request.body.secondary.length - 1) ? "PKTextAlignmentRight" : "PKTextAlignmentLeft",
+                });
+            }
+            });
+
+            request.body.auxiliary.forEach((field, index) => {
+                if (field.label !== '' || field.value !== '') {
+                    newPass.auxiliaryFields.push({
+                        key: `auxiliary${index}`,
+                        label: field.label,
+                        value: field.value,
+                        textAlignment: (index === request.body.secondary.length - 2 || index === request.body.secondary.length - 1) ? "PKTextAlignmentRight" : "PKTextAlignmentLeft",
+                    });
+                }
+                });
+
+            if (!request.body.codeAlt || request.body.codeAlt.trim() === "") {
+                newPass.setBarcodes({
+                    message: request.body.qrText,
+                    format: `PKBarcodeFormat${request.body.codeType}`,
+                    messageEncoding: "iso-8859-1",
+                });
+            } else {
+                newPass.setBarcodes({
+                    message: request.body.qrText,
+                    format: `PKBarcodeFormat${request.body.codeType}`,
+                    messageEncoding: "iso-8859-1",
+                    altText: request.body.codeAlt,
+                });
+            }
+            
+            // Downloading thumbnail and logo files from Firebase Storage and adding to pass
+            if (currentPassType == "generic" || currentPassType == "eventTicket") {
+                const thumbnailFile = request.body.thumbnailFile
+                const tempPath1 = path.join(os.tmpdir(), thumbnailFile)
+                try {
+                    await storageRef.file(`thumbnails/${thumbnailFile}`).download({destination: tempPath1})
+                } catch (error) {
+                    //
+                }
+                let buffer = Buffer.alloc(0);
+                try {
+                    buffer = fs.readFileSync(tempPath1)
+                } catch (error) {
+                    //
+                }
+                newPass.addBuffer("thumbnail.png", buffer)
+                newPass.addBuffer("thumbnail@2x.png", buffer)
```

Comment:
I honestly do not understand what are you doing here: creating an empty buffer, and then ditching it for a file read from the FileSystem. Why?
I mean, okay for the fallback, but assigning an empty buffer would be better if done only in the catch.

Also, what I do not understand is that `tempPath1` is something that comes from a Firebase API... so using `fs` wouldn't make it crash or something when using this code on a Cloud Function, just as it happened on the scripapi's video?

Reply:
Ah I think I remember why I did that now, I think I declared the buffer and assigned it to be empty outside the try catch so it could then still be accessed in the `newPass.addBuffer()` lines? Unless there is a way to assign the buffer inside the try catch and still access it outside, or simply do the `newPass.addBuffer()` inside the try catch, in both cases? Let me know what you think is best, again really sorry if it's a trivial question!

Reply:
The crash I refer to was about Scriptpapi writing on `fs` instead of Firebase and deploying the cloud function. It wasn't able to generate successfully a pass.

Let me ask you: you tried this on cloud or just locally? I think that `fs` is actually crashing but you are catching and losing the error if you are running this on the cloud.

In my opinion we should remove all the `fs` references and convert them to Firebase API.

Reply:
Hey, sorry for the delayed reply. I have tried this both locally using a firebase emulator, and the cloud, and I haven't seen any crashing (with the console logs put in), but you're right that might be a good idea generally. Do you have any idea what the firebase way to do it is?

Reply:
I guess I was wrong about `fs`. Maybe it is okay reading... but not writing? I mean, here it is where scriptpapi shows the issue: https://youtu.be/rJZdPoXHtzI?t=2178.

So it is either sync writing or writing at all. Or maybe it is all about temp writing. I don't know.

I'd, maybe, use Cloud Storage for keeping files. If Cloud Functions work just like AWS Lambdas, I think that having files in the cloud folder, might increase the cold start function amount of time, as the "orchestrator" behind the scenes must clone or unzip or whatever all the files.